crates/registry/src/error.rs

```diff
@@ -7,9 +7,21 @@ pub enum RegistryError {
     #[error("missing version `{0}` on package `${0}`")]
     MissingVersionRelease(String, String),
     #[error("network error while downloading `${0}`")]
-    Network(String),
-    #[error("filesystem error: `{0}`")]
-    FileSystem(String),
+    Network(reqwest::Error),
+    #[error("io error `${0}`")]
+    Io(std::io::Error),
     #[error("serialization failed: `${0}")]
     Serialization(String),
 }
+
+impl From<std::io::Error> for RegistryError {
```

Comment:
Could we use `#[from]` here instead of manual From impls?

Reply:
oh possibly! i didn't know that existed, actually. I'll look into it.